### PR TITLE
[Event Hubs] Support for Batch Processing

### PR DESCRIPTION
**Pull request by jsquire**
# Summary 
The focus of these changes is to expose the ability to create a checkpoint with the `EventProcessorClient` as a protected member so that it can be used when overriding  `OnProcessingEventBatch` to work with batches rather than individual events in the handler.  Previously, creating a checkpoint was bound to the event arguments passed to the handler and required an awkward and complicated pattern to use.  By making this protected, the `EventProcessorClient` can be extended in a manner similar to `EventProcessor<T>` and allows working with batches without the need to write a custom storage implementation.

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs [hunk 1]
```diff
@@ -712,57 +712,55 @@ protected EventProcessorClient() : base()
...
                     EventHubName = EventHubName,
                     ConsumerGroup = ConsumerGroup,
-                    PartitionId = context.PartitionId,
-                    StartingPosition = EventPosition.FromOffset(eventData.Offset)
+                    PartitionId = partitionId,
+                    StartingPosition = EventPosition.FromOffset(eventData.Offset, false)
                 };
 
-                return StorageManager.UpdateCheckpointAsync(checkpoint, eventData, cancellationToken);
+                await StorageManager.UpdateCheckpointAsync(checkpoint, eventData, cancellationToken).ConfigureAwait(false);
```
Comment (christothes) [inline sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs:749]: just curious - why the change to await the task?

**Review by christothes — APPROVED** [review]

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs (hunk 1 above)
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs:749]: The API was internal and we didn't feel the interim member was interesting in the stack trace and wouldn't surface directly to callers, but now that it is protected, I wanted to snap to our and prefer awaiting since I don't have strong justification to not do so.

**Comment by jsquire** [comment]
Stalling this for now; after talking to Krzysztof, we're going to explore the possibility of exposing a storage provider abstract and the Blob Storage implementation here for use with `EventProcessor<T>`.

**Comment by jsquire** [comment]
This pull request is now obsolete due to checkpoint store support.
